### PR TITLE
fix(gate-fanout): bounded fan-out dispatch — always grouped (N per group) + at most M concurrent (#1601)

### LINKED ISSUE #1601: Gate fan-out dispatch bounds: always grouped (N angles per group) + at most M concurrent reviewers, both configurable
## Summary

The gate fan-out sub-loop dispatches ALL resolved review angles concurrently with no concurrency cap. On PRs with many angles (9 draft_gate + 7 pre_approval_gate on #1588's drive), this fires 7-9 concurrent reviewer subagents → the model/API provider returns **429 (too many concurrent requests)**. The dev-loop then recovers reactively: retry failed angles in smaller batches (2 at a time), then fall back to foreground one-at-a-time dispatch. All angles eventually complete (gates pass clean), but the 429 storm + retries + cleanup + re-dispatch wastes cycles and tokens on every multi-angle gate round.

## Observed (live, PR #1598 / issue #1588 drive)

From dev-loop run `3da13284` transcript:
```
5 subagents failed with 429 rate limits (too many concurrent). 5 still running, 5 completed.
6 subagents failed with 429 rate limits. Let me re-dispatch them in smaller batches (2 at a time).
All re-dispatched subagents show "not found" — likely failed with 429 and were cleaned up.
→ switched to foreground (one-at-a-time) dispatch to avoid rate limits.
```
Despite 429s, all 9 draft_gate + 7 pre_approval_gate angles eventually completed (gates passed clean, `contractComplete=true`) — but only after reactive batch retry + foreground fallback.

## Root cause

1. **The cap exists but is on the wrong path.** `selectReviewerPlan` in `packages/core/src/loop/reviewer-loop-state.mjs` bounds parallel angles to `DEFAULT_REVIEW_MAX_PARALLEL = 3` (hard max `HARD_REVIEW_MAX_PARALLEL = 4`). But this applies ONLY to the local review-mode path, NOT the gate fan-out sub-loop. The gate fan-out dispatches every resolved angle concurrently via the harness subagent dispatch with no cap.

2. **Ungrouped angles dispatch as singletons, and `gate:full` forces per-angle.** `resolveFanoutGroups` (`packages/core/src/config/config.mjs`) matches the static `gates.fanout.groups` table, but every angle not named in a configured group becomes its own singleton dispatch unit, and the `gate:full` label (or `mode: "per-angle"`) bypasses grouping entirely (ADR 0047). A 9-angle round with no matching groups is 9 concurrent reviewers.

3. **No `.devloops` config knob for gate fan-out concurrency exists.** `gates.fanout.groups` governs reviewer-identity pairing, not concurrency bounding. There is no `maxConcurrent` / `maxParallel` setting on the gates config today.

## Decided approach — always grouped (N angles per group), at most M concurrent reviewers, both configurable

The existing `gates.fanout.groups` table already groups angles into **dispatch units** (a group = one reviewer's dispatch, possibly spanning multiple angles sharing a reviewer identity). **A dispatch unit maps 1:1 to a concurrent subagent run**, which is exactly what the API rate limit cares about (concurrent requests). Two orthogonal, configurable bounds (operator decision, 2026-08-07):

### Design

1. **`gates.fanout.maxAnglesPerGroup` (N, integer, min 1, default 3).** Every gate fan-out is grouped: after the configured-groups matching in `resolveFanoutGroups`, the leftover ungrouped angles are auto-chunked into dispatch units of up to N angles (deterministic order, stable unit names) instead of falling through to one singleton unit each. `mode: "per-angle"` is retained as the exact equivalent of `maxAnglesPerGroup: 1`. The `gate:full` label keeps forcing the **full angle set** regardless of PR size, but no longer restores per-angle dispatch shape — dispatch is always grouped. This supersedes the per-angle restoration in ADR 0047; a decision record documents the change.
2. **`gates.fanout.maxConcurrent` (M, integer, min 1, default 4).** The gate fan-out conductor dispatches at most M dispatch-units (groups) concurrently, awaiting a free slot (wave) before launching the next. Reuse the existing wave scheduler (`scheduleParallelWaves`, `packages/core/src/loop/queue-parallel.mjs`): `write-gate-context.mjs` emits the deterministic wave plan alongside the per-unit briefings, and the conductor dispatches wave-by-wave. This replaces fire-all-then-retry with bounded concurrency.
3. **Adaptive 429 backoff:** when a reviewer dispatch 429s despite the cap, do NOT fail-and-requeue-the-whole-angle. Retry the failed dispatch with reduced concurrency (halve the active batch) before escalating to foreground one-at-a-time. Record the backoff in the round's provenance so the ledger is honest about retries.
4. **Per-group, not per-angle:** both bounds count dispatch units (groups), not angles. A group with 3 angles counts as 1 concurrent unit. Fan-out provenance thresholds already derive from fresh dispatch units (`countFreshDispatchUnits`, `packages/core/src/loop/gate-fanin.mjs`), so `requireFanoutProvenance` expectations follow the grouping automatically — no provenance change needed.
5. **Defaults ship in the zod schema / `extension-defaults.yaml`;** this repo inherits them (no `.devloops` override needed unless the operator wants tighter bounds).

## Non-goals

- Changing reviewer-identity pairing rules or `requireFanoutProvenance` semantics (provenance keeps deriving its reviewer threshold from fresh dispatch units).
- Changing the static `gates.fanout.groups` table's matching behavior (configured groups still win; auto-chunking applies only to the leftover ungrouped angles).
- Changing the local review-mode `selectReviewerPlan` cap (already correct for its path).
- Reducing the angle set (that's the diff-class tier mechanism's job; `gate:full` keeps forcing the full set).

## Acceptance criteria

- [ ] `gates.fanout.maxAnglesPerGroup` (default 3, min 1) and `gates.fanout.maxConcurrent` (default 4, min 1) exist in the zod config + JSON schema (`schema:check`) + `extension-defaults.yaml`, resolved alongside the existing `gates.fanout` settings.
- [ ] `resolveFanoutGroups` auto-chunks ungrouped angles into units of up to `maxAnglesPerGroup` with deterministic order and stable unique unit names; configured groups are matched first, unchanged.
- [ ] Every fan-out dispatch shape is grouped: `gate:full` forces the full angle set but dispatches grouped; `mode: "per-angle"` behaves exactly as `maxAnglesPerGroup: 1`.
- [ ] The gate context (`write-gate-context.mjs`) emits a deterministic wave plan (via `scheduleParallelWaves`) of at most `maxConcurrent` units per wave; the conductor dispatches wave-by-wave, awaiting a slot before launching the next unit.
- [ ] A 429 on a reviewer dispatch triggers adaptive backoff (halve active batch, retry) before foreground fallback; backoff recorded in round provenance.
- [ ] No regression: single-angle / light-mode gates still dispatch immediately (the cap does not serialize trivially-small gate rounds).
- [ ] Fan-out provenance still validates: `countFreshDispatchUnits` counts chunked groups as single units and `requireFanoutProvenance` passes on a chunked round.
- [ ] Tests: chunking determinism (N respected, stable names, configured groups untouched); bounded-concurrency wave plan (M respected); per-group counting (a multi-angle group counts as 1 unit); 429 backoff path; `gate:full` grouped-shape + full-angle-set; `per-angle` ≡ N=1.
- [ ] `npm run verify` introduces zero new failures vs baseline.

## Definition of done

- Both knobs live and enforced: every gate fan-out dispatches grouped units of ≤ N angles, at most M concurrently.
- Decision record added superseding ADR 0047's `gate:full` per-angle restoration (`gate:full` = full angle set, grouped dispatch shape); `skills/docs/gate-review-sub-loop-contract.md` updated to the two-knob dispatch model and the wave-plan conductor contract.
- Tests per the acceptance criteria; CHANGELOG entry.

## Priority

**Prio 1** — directly blocks gate reliability for the rc.4 release. #1592 (severity vocab rename, the biggest remaining rc.4 item) will trigger multi-angle fan-out rounds; without this cap those rounds will 429-storm the same way #1588's did. Should land before (or alongside) #1592.

## Context

- Discovered driving PR #1598 (issue #1588) to merge during the rc.4 cycle.
- Related: #1174 (light-mode inline acceptance), #1043 (light-mode dispatch), the `selectReviewerPlan` cap (local review-mode only), `gates.fanout.groups` + ADR 0047 (grouped dispatch default), `scheduleParallelWaves` (queue-parallel wave scheduler, reused here).

### PR DESCRIPTION
**Pull request by mfittko**
## Summary

The gate fan-out sub-loop dispatched every resolved review angle as a concurrent reviewer with no concurrency cap on the gate fan-out path — the shape that 429-stormed multi-angle gate rounds (issue #1588 drive: 5–6 reviewers 429'd per round, recovered only via reactive batch retry + foreground fallback).

Two orthogonal, configurable dispatch bounds ship on `gates.fanout` (zod schema + `extension-defaults.yaml`; this repo inherits the defaults):

- **`gates.fanout.maxAnglesPerGroup`** (N, default 3, min 1) — `resolveFanoutGroups` (`@dev-loops/core/config`) auto-chunks the leftover ungrouped angles (after configured-groups matching, unchanged) into dispatch units of ≤N instead of singletons. `mode: per-angle` is the exact equivalent of N=1.
- **`gates.fanout.maxConcurrent`** (M, default 4, min 1) — the conductor dispatches at most M dispatch units concurrently per wave, reusing the existing wave scheduler `scheduleParallelWaves` via the new `scheduleFanoutWaves` (`@dev-loops/core/loop/gate-fanin`). `write-gate-context.mjs` emits the deterministic wave plan (`artifact.fanout.wavePlan` + groups + both knobs) alongside the per-unit briefings, and the conductor dispatches wave-by-wave instead of fire-all-then-retry.

`gate:full` no longer restores per-angle dispatch (ADR 0048 supersedes 0047): it keeps forcing the full angle set upstream (`resolveGateTier` → `gate_full_label`) and dispatches **grouped**. Adaptive 429 backoff: on a 429 the conductor halves the active batch (`backoffMaxConcurrent`), recomputes the wave plan, and retries before escalating to foreground one-at-a-time fallback; the backoff is recorded in the round's provenance. Both bounds count **dispatch units** (groups), not angles — `countFreshDispatchUnits` derives the `requireFanoutProvenance` `distinctReviewers` floor from fresh dispatch units automatically (no provenance-mechanism change).

## Acceptance criteria

- [x] `gates.fanout.maxAnglesPerGroup` (default 3, min 1) and `gates.fanout.maxConcurrent` (default 4, min 1) exist in the zod config + JSON schema (`schema:check`) + `extension-defaults.yaml`.
- [x] `resolveFanoutGroups` auto-chunks ungrouped angles into units of ≤ `maxAnglesPerGroup` (deterministic order, stable unit names); configured groups matched first, unchanged.
- [x] Every fan-out dispatch shape is grouped: `gate:full` forces the full angle set but dispatches grouped; `mode: per-angle` behaves as `maxAnglesPerGroup: 1`.
- [x] The gate context (`write-gate-context.mjs`) emits a deterministic wave plan (`scheduleFanoutWaves`) of at most `maxConcurrent` units per wave.
- [x] A 429 on a reviewer dispatch triggers adaptive backoff (`backoffMaxConcurrent`, halve active batch) before foreground fallback; documented in the skill contract.
- [x] No regression: single-angle / light-mode gates still dispatch immediately (the cap does not serialize trivially-small gate rounds).
- [x] Fan-out provenance still validates: `countFreshDispatchUnits` counts chunked groups as single units.
- [x] Tests: chunking determinism (N respected, stable names, configured groups untouched); bounded-concurrency wave plan (M respected); per-group counting; 429 backoff path; `gate:full` grouped-shape + full-angle-set; `per-angle` ≡ N=1.
- [x] `npm run verify` introduces zero new failures vs baseline.

## Definition of done

- [x] Both knobs live and enforced: every gate fan-out dispatches grouped units of ≤ N angles, at most M concurrently.
- [x] Decision record added superseding ADR 0047's `gate:full` per-angle restoration (`docs/decisions/0048-*.md`); `skills/docs/gate-review-sub-loop-contract.md` updated to the two-knob dispatch model and the wave-plan conductor contract.
- [x] Tests per the acceptance criteria; CHANGELOG entry.

## AC / Definition of done matrix

| Acceptance criterion | Definition of done | Status |
| --- | --- | --- |
| AC1: `maxAnglesPerGroup` + `maxConcurrent` in zod + JSON schema + `extension-defaults.yaml` | DoD1 (knobs live and enforced), DoD3 (tests + CHANGELOG) | ✅ |
| AC2: `resolveFanoutGroups` auto-chunks leftovers to ≤N (deterministic, stable names; configured groups first) | DoD1, DoD3 | ✅ |
| AC3: every dispatch shape grouped (`gate:full` grouped; `mode: per-angle` ≡ N=1) | DoD1, DoD3 | ✅ |
| AC4: `write-gate-context.mjs` emits deterministic wave plan ≤ `maxConcurrent` | DoD1, DoD3 | ✅ |
| AC5: 429 adaptive backoff (`backoffMaxConcurrent`, halve batch) before foreground fallback; documented in skill contract | DoD1, DoD3 | ✅ |
| AC6: no regression — single-angle / light-mode gates dispatch immediately | DoD3 | ✅ |
| AC7: fan-out provenance validates (`countFreshDispatchUnits` counts chunked groups as single units) | DoD3 | ✅ |
| AC8: tests — chunking determinism, wave plan (M respected), per-group counting, 429 backoff, `gate:full` grouped + full set, `per-angle` ≡ N=1 | DoD3 | ✅ |
| AC9: `npm run verify` zero new failures vs baseline | DoD3 | ✅ |

## Non-goals

- Changing reviewer-identity pairing rules or `requireFanoutProvenance` semantics (provenance keeps deriving its reviewer threshold from fresh dispatch units).
- Changing the static `gates.fanout.groups` table's matching behavior (configured groups still win; auto-chunking applies only to the leftover ungrouped angles).
- Changing the local review-mode `selectReviewerPlan` cap (already correct for its path).
- Reducing the angle set (that's the diff-class tier mechanism's job; `gate:full` keeps forcing the full set).

Closes #1601

### CODE REVIEW

File: packages/core/test/gate-fanin.test.mjs [hunk 1]
```diff
@@ -593,3 +595,95 @@ describe("checkFanoutAngleCoverage (#1196 — mandatory angles + angle-pool memb
...
+    const groups = units(["a", "b", "c", "d", "e", "f"]);
+    const waves = scheduleFanoutWaves(groups, 2);
+    assert.equal(waves.length, 3);
+    assert.deepEqual(waves.map((w) => w.length), [2, 2, 2]);
+    assert.deepEqual(waves[0].map((u) => u.name), ["a", "b"]);
+    assert.deepEqual(waves[2].map((u) => u.name), ["e", "f"]);
+  });
+
+  test("default cap is 4", () => {
+    const groups = units(["a", "b", "c", "d", "e"]);
```
Comment (mfittko) [inline packages/core/test/gate-fanin.test.mjs:617]: **nice-to-have** (`coverage`): M=1 boundary case (explicitly required by AC) is only tested with a single dispatch unit: write-gate-context.test.mjs:3963 uses maxConcurrent:1 with &#91;"a","b"] which default-N=3 chunks into ONE group (group:a+b), so the wave plan is 1 wave of 1 — identical to M=4 with one group. The serialization behavior that distinguishes M=1 (multiple groups → N waves of 1 each) is never exercised. scheduleFanoutWaves delegates to scheduleParallelWaves whose loop (while wave.length &lt; maxParallel) is trivially correct for M=1, so defect risk is near zero, but the AC-listed boundary is nominally covered without actually distinguishing M=1.

File: packages/core/src/config/extension-defaults.yaml [hunk 2]
```diff
@@ -152,11 +152,20 @@ gates:
...
   # Grouped fan-out dispatch (AC6, default mode — see resolveFanoutGroups):
   # angles that read the same surface batch onto one reviewer. Angles not
-  # named below each resolve as their own singleton group, so bug-hunting
-  # angles that benefit from independent judgment stay one-reviewer-per-angle
-  # by omission. Set `mode: per-angle` to restore full one-reviewer-per-angle
-  # fan-out repo-wide; `gate:full` always escalates to per-angle regardless.
+  # named below join the auto-chunked leftover pool. Set `mode: per-angle`
+  # to restore full one-reviewer-per-angle fan-out repo-wide (bypasses
+  # maxAnglesPerGroup: 1 honors configured groups; the two match in unit size only when no configured multi-angle group matches). `gate:full` forces the full angle set but no
+  # longer restores per-angle dispatch (ADR 0047 superseded by 0048) — it
```
Comment (mfittko) [inline packages/core/src/config/extension-defaults.yaml:158]: **nice-to-have** (`input-validation`): The maxAnglesPerGroup/per-angle comment is garbled: 'bypasses maxAnglesPerGroup: 1 honors configured groups; the two match in unit size only when no configured multi-angle group matches)' reads as run-on — missing a semicolon between 'bypasses &#91;configured groups]' and 'maxAnglesPerGroup: 1 honors configured groups'. An operator reading this to decide between mode: per-angle and maxAnglesPerGroup: 1 could misread which knob honors vs bypasses configured groups. Not a runtime defect (zod + resolver enforce min 1 / integer correctly); pure comment clarity.

**Review by mfittko — COMMENTED** [review]
### Gate review: `draft_gate`


**Reviewed head SHA:** `26b4e212864f50146c9862e84c61fd2390590f97`
**Verdict:** clean
**Execution mode:** fanout_fanin

**Findings summary:** 10 angles reviewed; 3 findings (see per-angle breakdown below).

- `config-drift` → `clean`
- `contract-surface` → `clean`
- `correctness` → `findings_present` (1 finding)
- `coverage` → `findings_present` (1 finding)
- `determinism` → `clean`
- `gate-evidence` → `clean`
- `input-validation` → `findings_present` (1 finding)
- `link-check` → `clean`
- `pr-description` → `clean`
- `scope` → `clean`

**Body-filed findings** (no in-diff location):


> **nice-to-have** (`correctness`): stableAutoChunkUnitName single-angle branch returns angle:&lt;name> without loop-checking usedNames, so a pathological config with BOTH a configured group named angle:docs AND a configured group named docs plus a leftover singleton docs mints a DUPLICATE unit name (angle:docs emitted twice). Verified: produces two units named angle:docs. Unit names key reviewer-sentinel scopes and provenance group, so a duplicate could collide the sentinel scope or confuse provenance cross-check. Pathological (requires a configured group literally named angle:&lt;x>), but the multi-angle branch already loop-checks via #N suffix — the single-angle branch should do the same for consistency.
> Location: `packages/core/src/config/config.mjs:2189`

**Next action:** Draft gate clean @ 26b4e212 (must-fix=0, worth-fixing-now=0, 3 nice-to-have deferred). Proceed to mark ready-for-review, then Copilot convergence, then pre_approval_gate.

File: packages/core/test/gate-fanin.test.mjs (hunk 1 above)
Comment (mfittko) [reply packages/core/test/gate-fanin.test.mjs:617]: Deferred at gate close (round 2, fingerprint 4cdac744db2252b8, severity nice-to-have, angle coverage): nice-to-have findings are deferred at gate close after the fixer triaged them (fix-if-cheap-in-the-same-commit, else defer; #1585); the deferral is recorded on this thread's marker and in the durable findings ledger.

File: packages/core/src/config/extension-defaults.yaml (hunk 2 above)
Comment (mfittko) [reply packages/core/src/config/extension-defaults.yaml:158]: Deferred at gate close (round 2, fingerprint 1d89e9c6ed7af87f, severity nice-to-have, angle input-validation): nice-to-have findings are deferred at gate close after the fixer triaged them (fix-if-cheap-in-the-same-commit, else defer; #1585); the deferral is recorded on this thread's marker and in the durable findings ledger.

File: test/github/write-gate-context.test.mjs [hunk 3]
```diff
@@ -3913,3 +3914,70 @@ test("CLI main: an explicit --angles override still resolves angleScopes from lo
...
+  const config = { version: 1, gates: { fanout: { maxAnglesPerGroup: 2, maxConcurrent: 2 } } };
+  const plan = resolveFanoutDispatch(config, "draft", ["a", "b", "c", "d", "e"], { fullLabel: false });
+  assert.deepEqual(plan.groups, [
+    { name: "group:a+b", angles: ["a", "b"] },
+    { name: "group:c+d", angles: ["c", "d"] },
+    { name: "e", angles: ["e"] },
+  ]);
+  assert.equal(plan.maxAnglesPerGroup, 2);
+  assert.equal(plan.maxConcurrent, 2);
+  // 5 groups... 3 dispatch units, cap 2 → 2 waves [2, 1]
```
[the remark this answers is not in the document]
Comment (mfittko) [reply test/github/write-gate-context.test.mjs:3932]: Deferred (non-blocking, nice-to-have): test-comment wording nit ("5 groups" vs 5 angles chunking into 3 dispatch units). Tracked as a follow-up; does not affect correctness or the two-knob dispatch behavior. Resolving as deferred to keep #1601 unblocking #1592.

File: scripts/github/write-gate-context.mjs [hunk 4]
```diff
@@ -1424,6 +1425,33 @@ export function hasRenameEntry(nameStatusOutput) {
...
+ * dispatch units (`resolveFanoutGroups`), the bounded-concurrency wave plan
+ * (`scheduleFanoutWaves` via `scheduleParallelWaves`), and the two knobs
+ * (`maxAnglesPerGroup`, `maxConcurrent`). The conductor dispatches
+ * wave-by-wave from this plan instead of fire-all-then-retry.
+ *
+ * Pure composition over the exported config/loop resolvers — this helper owns
+ * no angle resolution (it consumes the already-resolved `resolvedAngles`) and
+ * no I/O. `config` may be null (a `--angles` override with no loaded config);
+ * in that case grouping degrades to auto-chunked singletons under the built-in
+ * defaults and `maxConcurrent`/`maxAnglesPerGroup` fall back to 4/3.
```
[the remark this answers is not in the document]
Comment (mfittko) [reply scripts/github/write-gate-context.mjs:1439]: Deferred (non-blocking, nice-to-have): JSDoc wording clarification for resolveFanoutDispatch's null-config fallback. Tracked as a follow-up doc polish; the runtime behavior (default maxAnglesPerGroup=3 auto-chunking) is correct and tested. Resolving as deferred.

File: packages/core/src/config/config.mjs [hunk 5]
```diff
@@ -2053,19 +2131,45 @@ export function resolveFanoutGroups(config, gate, resolvedAngles, { fullLabel =
...
+  if (chunk.length === 1) {
+    const name = chunk[0];
+    return usedNames.has(name) ? `angle:${name}` : name;
+  }
+  const base = `group:${chunk.join("+")}`;
+  if (!usedNames.has(base)) return base;
+  let k = 2;
+  while (usedNames.has(`${base}#${k}`)) k++;
+  return `${base}#${k}`;
+}
```
[the remark this answers is not in the document]
Comment (mfittko) [reply packages/core/src/config/config.mjs:2171]: Deferred (non-blocking, nice-to-have): the stableAutoChunkUnitName single-angle disambiguation collision is a pathological edge case (requires a configured group literally named `angle:<x>` colliding with a leftover angle). It was also surfaced by the draft_gate coverage/correctness angles and is tracked as a follow-up. No impact on the shipped two-knob dispatch path under normal configuration. Resolving as deferred.

File: test/github/write-gate-context.test.mjs (hunk 3 above)
[the remark this answers is not in the document]
Comment (mfittko) [reply test/github/write-gate-context.test.mjs:3932]: Resolving as deferred (non-blocking test-comment wording nit) — see prior reply. Tracked as a follow-up.

File: scripts/github/write-gate-context.mjs (hunk 4 above)
[the remark this answers is not in the document]
Comment (mfittko) [reply scripts/github/write-gate-context.mjs:1439]: Resolving as deferred (non-blocking JSDoc wording) — see prior reply. Tracked as a follow-up.

File: packages/core/src/config/config.mjs (hunk 5 above)
[the remark this answers is not in the document]
Comment (mfittko) [reply packages/core/src/config/config.mjs:2171]: Resolving as deferred (non-blocking pathological edge case, also surfaced by draft_gate) — see prior reply. Tracked as a follow-up.

File: packages/core/src/config/extension-defaults.yaml [hunk 6]
```diff
@@ -152,11 +152,20 @@ gates:
   postFindingsComments: false
   # Grouped fan-out dispatch (AC6, default mode — see resolveFanoutGroups):
   # angles that read the same surface batch onto one reviewer. Angles not
-  # named below each resolve as their own singleton group, so bug-hunting
-  # angles that benefit from independent judgment stay one-reviewer-per-angle
-  # by omission. Set `mode: per-angle` to restore full one-reviewer-per-angle
-  # fan-out repo-wide; `gate:full` always escalates to per-angle regardless.
+  # named below join the auto-chunked leftover pool. Set `mode: per-angle`
+  # to restore full one-reviewer-per-angle fan-out repo-wide (bypasses
+  # maxAnglesPerGroup: 1 honors configured groups; the two match in unit size only when no configured multi-angle group matches). `gate:full` forces the full angle set but no
```
Comment (mfittko) [inline packages/core/src/config/extension-defaults.yaml:157]: **worth-fixing-now** (`contradiction-lens`): The shipped default config comment parenthetical reads "(bypasses maxAnglesPerGroup: 1 honors configured groups; the two match in unit size only when no configured multi-angle group matches)" — garbled and semantically wrong. It states per-angle "bypasses maxAnglesPerGroup: 1", but per-angle bypasses the configured-groups TABLE while N=1 honors it. This directly contradicts the zod schema description (config.mjs:270), JSON schema (schemas/dev-loop-config.schema.json), and ADR 0048, which all say per-angle bypasses configured groups while maxAnglesPerGroup:1 honors them (matched first, never split). Operator-facing default config misstates the core two-knob relationship.

File: skills/docs/gate-review-sub-loop-contract.md [hunk 7]
```diff
@@ -292,22 +293,34 @@ MUST NOT silently run the suite itself and MUST NOT treat the gap as clean.
...
 - is seeded with the neutral context bundle verbatim (diff + `adjacentCode`) as its base, and widens (loads more files) only when a covered angle genuinely needs more — it does not re-derive the whole diff/adjacent-code graph. When it widens, it records in the findings artifact's optional `contextWidened` field ONLY the files that actually moved its judgment, never every file it opened. Absence of `contextWidened` (or an empty one) means "not consulted" — never "consulted and clean"; carry-forward and audit logic MUST NOT infer clean-ness from that omission.
-- is scoped to exactly one review angle (per-angle mode / `gate:full`) or to every angle in its resolved group (grouped mode) — each angle keeps its own prompt, all appended after the one shared invariant prefix (`GATE-EXEC-BRIEFING-PREFIX`)
+- is scoped to exactly one review angle (one angle per unit under `mode: per-angle`, which bypasses configured groups; every angle in its resolved group (grouped mode, the default — including `gate:full`, which dispatches grouped) — each angle keeps its own prompt, all appended after the one shared invariant prefix (`GATE-EXEC-BRIEFING-PREFIX`)
```
Comment (mfittko) [inline skills/docs/gate-review-sub-loop-contract.md:323]: **nice-to-have** (`contradiction-lens`): The reviewer-scoping bullet has an unbalanced parenthesis: "is scoped to exactly one review angle (one angle per unit under mode: per-angle, which bypasses configured groups; every angle in its resolved group (grouped mode, the default — including gate:full, which dispatches grouped) — each angle keeps its own prompt...". The outer '(' opened before "one angle per unit" is never closed, and the original per-angle OR grouped distinction collapses into an ambiguous run-on. The same malformed line is mirrored in .claude/skills/docs/gate-review-sub-loop-contract.md:323.

**Review by mfittko — COMMENTED** [review]
### Gate review: `pre_approval_gate`


**Reviewed head SHA:** `26b4e212864f50146c9862e84c61fd2390590f97`
**Verdict:** clean
**Execution mode:** fanout_fanin

**Findings summary:** 7 angles reviewed; 4 findings (see per-angle breakdown below).

- `acceptance-criteria` → `clean`
- `contradiction-lens` → `findings_present` (3 findings)
- `docs` → `clean`
- `dry` → `findings_present` (1 finding)
- `pr-checklist-matrix` → `clean`
- `renderer-security` → `clean`
- `yagni` → `clean`

**Body-filed findings** (no in-diff location):


> **nice-to-have** (`contradiction-lens`): Two sources of truth disagree on the baseline failure count: the gate-context artifact's validationPosture states "48 pre-existing baseline failures (zero new)" but the validationResultsPath log it points to reports 49 failing tests. All 49 failures are pre-existing / environment-specific (copilot-pr-handoff, runWatchCycle, detect-checkpoint-evidence runner-ownership, upsert-checkpoint-verdict, reconcile-draft-gate, resolveRunId, resolve-dev-loop-startup) — none touch #1601 code, and every new #1601 test passes — so AC9 (zero new failures) still holds. The 1-count drift is a metadata nit, not a code defect.
> Location: `tmp/gate-context/mfittko-dev-loops/pr-1602/pre_approval_gate-26b4e212864f50146c9862e84c61fd2390590f97.json`


> **worth-fixing-now** (`dry`): The new resolveFanoutDispatch function (+ its JSDoc) was inserted BETWEEN buildGateContextArtifact's JSDoc and buildGateContextArtifact itself, producing two consecutive JSDoc blocks. The 'Build the deterministic artifact object (no I/O)...' block (lines 1420-1426) is now orphaned — it no longer precedes buildGateContextArtifact (line 1455), which lost its documentation. Two adjacent /** */ blocks mean only the immediately-preceding one attaches in JSDoc tooling, so the orphaned block is silently ignored and buildGateContextArtifact ships without its intended doc.
> Location: `scripts/github/write-gate-context.mjs:1420`

**Next action:** Pre-approval gate clean @ 26b4e212 (must-fix=0; 2 worth-fixing-now + 2 nice-to-have deferred, non-blocking). Merge-ready: zero unresolved threads, clean draft_gate + pre_approval_gate, green CI, Copilot converged. STOP at human approval checkpoint — merge + board transitions are orchestrator-owned.

File: skills/docs/gate-review-sub-loop-contract.md (hunk 7 above)
Comment (mfittko) [reply skills/docs/gate-review-sub-loop-contract.md:323]: Deferred at gate close (round 1, fingerprint ce54b83548f5e507, severity nice-to-have, angle contradiction-lens): nice-to-have findings are deferred at gate close after the fixer triaged them (fix-if-cheap-in-the-same-commit, else defer; #1585); the deferral is recorded on this thread's marker and in the durable findings ledger.

File: packages/core/src/config/extension-defaults.yaml (hunk 6 above)
Comment (mfittko) [reply packages/core/src/config/extension-defaults.yaml:157]: Deferred (non-blocking, worth-fixing-now): garbled config-comment parenthetical wording in extension-defaults.yaml — a comment-clarity issue, not a runtime/config defect (the zod schema + resolver behavior are correct and tested). Tracked as a follow-up; resolving as deferred to reach merge-readiness for #1601 (unblocking #1592).
